doc/nixbuild.net.md

```diff
@@ -0,0 +1,96 @@
+
+[^system]: As of this writing, only for Linux on x86 and ARM; up-to-date list of systems [here](https://docs.nixbuild.net/supported-systems/index.html).
+
+Getting access
+---------------
+
+Pending at least the [initial prototype](https://input-output.atlassian.net/browse/PLT-4253) of the tribe-wide access control system, initial access for a user is a manual process. Please reach out to `Shea Levy` on Slack for access.
+
+**Note that the given tokens are secrets and should be managed as such!**
+
+Initial tokens will be set to expire around 2023-04-19, with the hope that by then an automated token server will exist. Use `biscuit inspect` (see [Managing access tokens](#managing-access-tokens)) to confirm the exact expiration.
+
+
+Setting up remote builders
+-------------------------
+
+nixbuild.net has documentation for setting everything up on [NixOS](https://docs.nixbuild.net/getting-started/#quick-nixos-configuration), or for configuring [SSH](https://docs.nixbuild.net/getting-started/#ssh-configuration) and [Nix](https://docs.nixbuild.net/getting-started/#nix-configuration) separately. From there, you can find links to more detailed documentation on remote builders. However, due to our usage of biscuit tokens instead of SSH keys changes to the SSH configuration are needed.
+
+Instead of an entry like:
+
+```
+Host eu.nixbuild.net
+  PubkeyAcceptedKeyTypes ssh-ed25519
+  IdentityFile /path/to/your/private/key
+```
+
+Your entry should look like:
+
+```
+# when connecting to eu.nixbuild.net
+Host eu.nixbuild.net
+  # don't use any authentication mechanisms SSH expects (password, SSH keys, etc.)
+  PreferredAuthentications none
+  # log in as the 'authtoken' user
+  # The token itself contains user information that nixbuild.net needs
+  User authtoken
+  # Send the 'token' environment variable from Nix to nixbuild.net in the SSH session.
+  SendEnv token
+  # Alternatively to SendEnv, you can have a line like:
+  #   SetEnv token=<token>
+  # This will mean you don't have to modify Nix's environment (see below).
+  # But this will involve putting your token, which is a secret, directly into
+  # a configuration file.
+```
+
+and then you should ensure that the environment of your Nix daemon (or, if you're running in single-user mode, your Nix build command) has the `token` variable set to the contents of your token.
+
+Managing access tokens
+----------------------
+We are using [Biscuit](biscuitsec.org/) tokens to access nixbuild.net. Among other things, these tokens allow for *offline attenuation*: If you have a token with a certain access, you can freely generate a new token with more restricted access without going through any central service. In this way, you can securely delegate capabilities; for example, given a token obtained as above, you could generate a new token that expires sooner (or, after future upgrades to nixbuild.net, can only spend a certain number of build hours per week) for use in CI.
```

Comment:
cool

Reply:
What's the best practice for e.g. CI tokens. Generate an attenuated token from a specific user's token? Get a specific one for the CI so it's not dependent on an individual?

Reply:
For now let's attenuate from specific users. I'm ambivalent on a good long-term practice... Operationally, it comes down to: When MPJ leaves the company (long may it be delayed!), should we have to reprovision credentials for all the services he set up? On the one hand, it seems like the answer should be no, because the teams and systems persist. On the other, if he created the creds he could have copied them, and if we say "we trust him not to have" why not equally trust him not to continue using his account access?

For this use case it's a bit academic, and I think we'll end up allocating project or team-wide tokens. But it's not an obvious question in principle to me.

Reply:
Right, I'm just looking for some concrete guidance as a user here. Otherwise I'm going to come bother you to ask :)

Reply:
Right, for now just use the token you get as a user, we'll figure this out when we have the automated oauth->token service set up.